### PR TITLE
Implement required connectionManagerService and mediaReceiverRegistrarService

### PR DESCRIPTION
**Pull request by tsynik**
this fix DLNA browse on LG TVs and under Windows 10 Explorer

PR also have some addons from rclone repo for Samsung TVs (some of them still have browse issues)

Probably fix for https://github.com/anacrolix/dms/issues/40 https://github.com/anacrolix/dms/issues/78 https://github.com/anacrolix/dms/issues/73

### CODE REVIEW

**Comment by anacrolix** [comment]
The formatting and imports are messed up. Try running `goimports -w .`. That leaves these failures:
```
$ go test ./...
# github.com/anacrolix/dms/dlna/dms [github.com/anacrolix/dms/dlna/dms.test]
dlna/dms/mrrs_test.go:21:38: undefined: testURL
dlna/dms/mrrs_test.go:21:69: undefined: mustMarshalXML
?   	github.com/anacrolix/dms	[no test files]
--- FAIL: TestContentFeaturesString (0.00s)
    dlna_test.go:14: DLNA.ORG_OP=10;DLNA.ORG_CI=1;DLNA.ORG_FLAGS=01700000000000000000000000000000
FAIL
FAIL	github.com/anacrolix/dms/dlna	0.117s
FAIL	github.com/anacrolix/dms/dlna/dms [build failed]
ok  	github.com/anacrolix/dms/misc	(cached)
?   	github.com/anacrolix/dms/play/termsig	[no test files]
?   	github.com/anacrolix/dms/rrcache	[no test files]
?   	github.com/anacrolix/dms/soap	[no test files]
?   	github.com/anacrolix/dms/ssdp	[no test files]
?   	github.com/anacrolix/dms/transcode	[no test files]
ok  	github.com/anacrolix/dms/upnp	(cached)
?   	github.com/anacrolix/dms/upnpav	[no test files]
FAIL
```

**Comment by tsynik** [comment]
> The formatting and imports are messed up. Try running `goimports -w .`.

This part was from rclone's https://github.com/rclone/rclone/blob/master/cmd/serve/dlna/dlna_test.go for new service, not checked it. sorry. it.s ok now:
ProBooK:dms nikk$ go test ./...
?   	github.com/anacrolix/dms	[no test files]
ok  	github.com/anacrolix/dms/dlna	0.011s
ok  	github.com/anacrolix/dms/dlna/dms	0.030s
ok  	github.com/anacrolix/dms/misc	0.011s
?   	github.com/anacrolix/dms/play/termsig	[no test files]
?   	github.com/anacrolix/dms/rrcache	[no test files]
?   	github.com/anacrolix/dms/soap	[no test files]
?   	github.com/anacrolix/dms/ssdp	[no test files]
?   	github.com/anacrolix/dms/transcode	[no test files]
ok  	github.com/anacrolix/dms/upnp	0.015s
?   	github.com/anacrolix/dms/upnpav	[no test files]

**Comment by tsynik** [comment]
> The formatting and imports are messed up

@anacrolix  do I need to apply gofmt from go1.17 also?
gofmt -l -w .
dlna/dms/dms_others.go
dlna/dms/dms_unix.go
dlna/dms/dms_unix_test.go
dlna/dms/dms_windows.go
play/attrs.go
play/bool.go
play/closure.go
play/execbug.go
play/execgood.go
play/ffprobe.go
play/mime.go
play/parse_http_version.go
play/print-ifs.go
play/scpd.go
play/soap.go
play/transcode.go
play/url.go

generates this: 
//go:build ignore
// +build ignore
etc

**Comment by anacrolix** [comment]
Yes please.

File: dlna/dms/cms-desc.go [hunk 1]
```diff
@@ -0,0 +1,185 @@
+package dms
```
Comment (anacrolix) [inline dlna/dms/cms-desc.go]: It might help to rename this file back to cm-service-desc.go, to reduce the diff.

File: dlna/dms/dms_others.go [hunk 2]
```diff
@@ -1,4 +1,5 @@
-//+build !linux,!darwin,!windows
+//go:build !linux && !darwin && !windows
```
Comment (anacrolix) [inline dlna/dms/dms_others.go]: I don't know why these diffs are showing, this is also in master now.

File: dlna/dms/cds-desc.go [hunk 3]
```diff
@@ -0,0 +1,506 @@
+package dms
```
Comment (anacrolix) [inline dlna/dms/cds-desc.go]: Same here we could rename back to cd-service-desc.go to reduce the diff.

**Comment by anacrolix** [comment]
Do we need any license notice from rclone?

**Comment by tsynik** [comment]
> Do we need any license notice from rclone?

here we don't have direct code from rclone - just similar logic.

File: dlna/dms/dms_others.go (hunk 2 above)
Comment (tsynik) [reply dlna/dms/dms_others.go]: it used , instead && - goimports vs gofmt

**Comment by rsteube** [comment]
Oh awesome, yes this fixes #78. Just tried it out and works great so far. Thanks for the work.

**Comment by anacrolix** [comment]
I mentioned on the probable fix issues to try this PR. Thanks for your great work @tsynik .

**Comment by anacrolix** [comment]
I'm considering squashing and merging this. If I can get some more feedback from users that it's working for them. Thanks again to @tsynik for the good work.

**Comment by anacrolix** [comment]
Okay let's give it a go. I'm not hearing any more feedback and you put a lot of effort in to this.

**Pull request merged**